Arelle/Arelle
# Handle List of Packages in Web Server

#### Reason for change
Resolves #1065

#### Description of change
* Make mypy type hint checking opt-out instead of opt-in for modules.
* Add type hints to CntlrWebMain module.
* Update the webserver to handle the list format of packages introduced by #940.
* Update webserver cntrl to use standard RuntimeOptions class.

#### Steps to Test
Verify no errors are thrown from:
* Start webserver with multiple packages:
`python arelleCmdLine.py --webserver=0.0.0.0:8080 --package https://filings.xbrl.org/549300PZMFIQR79Q0T97/2023-12-31/ESEF/IS/0/549300PZMFIQR79Q0T97-2023-12-31-en.zip --package https://filings.xbrl.org/2138005WRSDC4DI3BJ43/2023-12-31/ESEF/IS/0/2138005WRSDC4DI3BJ43-2023-12-31-is.zip`
* Use configure API to load other packages:
`http://localhost:8080/rest/configure?packages=https://filings.xbrl.org/549300D2K6PKKKXVNN73/2022-12-31/ESEF/DK/0/APMM-2022-12-31-en.zip|https://filings.xbrl.org/5299008YTLEN09WTHW26/2023-12-31/ESEF/IS/0/5299008YTLEN09WTHW26-2023-12-31.zip`
* Validate with even more packages:
`http://localhost:8080/rest/xbrl/validation?plugins=validate/ESEF&disclosureSystem=esef&media=json&file=https://filings.xbrl.org/2138001CNF45JP5XZK38/2023-12-31/ESEF/FI/1/KONE-2023-12-31-en.zip&packages=https://filings.xbrl.org/2138004PXX8LWGHGL872/2023-12-31/ESEF/DK/0/2138004PXX8LWGHGL872-2023-12-31-en.zip|https://filings.xbrl.org/5299007OWYZ6I1E46843/2023-12-31/ESEF/DK/0/5299007OWYZ6I1E46843-2023-12-31-en.zip`

**review**:
@Arelle/arelle
